# load libraries and data
```{r}
library(dplyr)
library(tidyr)
library(ggplot2)
library(forcats)
library(readxl)

# load eDNA assessment literature data & fix formatting
e1 <- read_xlsx("data/FCO_biodiversity_data_working.xlsx", sheet = "eDNA_assessment") %>%
  filter(pdf == "y", !is.na(region), !is.na(invertebrates))

etmp <- e1 %>% select(mammals:prokaryotes) %>% as.matrix()
etmp[etmp == "X"] <- 1 # change Xs to 1s
etmpnum <- matrix(as.numeric(etmp), ncol = ncol(etmp))
etmpnum[etmpnum > 0] <- 1 # change >1 to 1
colnames(etmpnum) <- colnames(etmp)

e <- e1 %>%
  select(-c(mammals:prokaryotes)) %>%
  bind_cols(etmpnum) %>%
  select(paperid:year, region:substrate, impact:numsamp, mammals:prokaryotes, target_noDNA, summary, title, journal, vol, num, pages) %>%
  rowwise() %>%
  mutate(numtaxgroups = sum(c_across(mammals:prokaryotes))) %>%
  ungroup() %>%
  mutate(othervert = case_when(mammals==1|birds==1|amphibians==1|reptiles==1 ~ 1,
                               TRUE ~ 0),
         vert = case_when(mammals==1|birds==1|amphibians==1|reptiles==1|fish==1 ~ 1,
                               TRUE ~ 0),
         metazoo = case_when(mammals==1|birds==1|amphibians==1|reptiles==1|fish==1|invertebrates==1 ~ 1,
                               TRUE ~ 0),
         microbe = case_when(prokaryotes==1|protists==1|fungi==1 ~ 1,
                               TRUE ~ 0),
           terr = case_when(grepl(x = hab_gen, pattern = "forest") ~ 1,
                            grepl(x = hab_gen, pattern = "terrestrial") ~ 1,
                            hab_gen == "cave" ~ 1,
                            hab_gen == "grassland" ~ 1,
                          TRUE ~ 0),
           forest = case_when(grepl(x = hab_gen, pattern = "forest") ~ 1,
                          TRUE ~ 0),
         year = as.numeric(year))


# load carbon offset project data & fix formatting
p1 <- read_xlsx("data/FCO_biodiversity_data_working.xlsx", sheet = "FCO_projects") #%>%
#   filter(!is.na(yearMR),
#          yearMR != "NA")

ptmp <- p1 %>% select(mammals:incidental_obs) %>% as.matrix()
ptmp[ptmp == "X"] <- 1 # change Xs to 1s
ptmpnum <- matrix(as.numeric(ptmp), ncol = ncol(ptmp))
ptmpnum[ptmpnum > 0] <- 1 # change >1 to 1
colnames(ptmpnum) <- colnames(ptmp)

p <- p1 %>%
  select(-c(mammals:incidental_obs)) %>%
  bind_cols(ptmpnum) %>%
  mutate(totn = as.numeric(camtrap) + as.numeric(formal_obs)) %>%
  rowwise() %>%
  mutate(numtaxgroups = sum(c_across(mammals:prokaryotes)),
         numanimalgroups = sum(c_across(c(mammals, birds, amphibians, 
                                          reptiles, fish, invertebrates))),
         nummethods = sum(c_across(camtrap:incidental_obs))) %>%
  ungroup() %>%
  filter(reg == "CCB")

pall <- p1 %>%
  select(-c(mammals:incidental_obs)) %>%
  bind_cols(ptmpnum) %>%
  mutate(totn = as.numeric(camtrap) + as.numeric(formal_obs)) %>%
  rowwise() %>%
  mutate(numtaxgroups = sum(c_across(mammals:prokaryotes)),
         numanimalgroups = sum(c_across(c(mammals, birds, amphibians, 
                                          reptiles, fish, invertebrates))),
         nummethods = sum(c_across(camtrap:incidental_obs))) %>%
  ungroup() %>%
  filter(ID != "1175",
         ID != "1641")
```
# taxa monitored
```{r}
# format data for eDNA assessments

e.taxplot <- e %>%
  select(mammals:prokaryotes) %>%
  pivot_longer(cols = mammals:prokaryotes) %>% 
  mutate(group = "eDNA assessments") %>%
  group_by(group, name) %>%
  summarize(num = sum(value, na.rm = T)) %>%
  ungroup() %>%
  mutate(name = fct_reorder(name, num),
         pct = 100*num/nrow(e)) 

e.taxplot.terr <- e %>%
  filter(hab_gen %in% c("forest", "grassland", "other terrestrial")) %>%
  select(mammals:prokaryotes) %>%
  pivot_longer(cols = mammals:prokaryotes) %>% 
  mutate(group = "eDNA assessments") %>%
  group_by(group, name) %>%
  summarize(num = sum(value, na.rm = T)) %>%
  ungroup() %>%
  mutate(name = fct_reorder(name, num),
         pct = 100*num/130) 

taxplot <- pall %>%
  select(mammals:prokaryotes) %>%
  pivot_longer(cols = mammals:prokaryotes) %>% 
  mutate(group = "FCO projects") %>%
  group_by(group, name) %>%
  summarize(num = sum(value, na.rm = T)) %>%
  ungroup() %>%
  mutate(name = fct_reorder(name, num),
         pct = 100*num/nrow(pall)) %>%
  bind_rows(e.taxplot) %>%
  mutate(groupord = c(rep(1,10),rep(2,10)),
         group = fct_reorder(group, groupord),
         imagew = pct * 125 / 77.1084337)
  
taxplot %>%
  ggplot() +
  geom_col(aes(x = pct, y = name)) +
  facet_wrap(~group) +
  labs(x = "% of projects or studies", y = "") +
  theme_bw() +
  theme(text = element_text(size = 14))
  
ggsave("figures/fig2_taxa_monitored_CCB+GS.png", dpi = 400, width = 6, height = 4)
```
# increasing representation of vertebrates over time?
```{r}

mod <- glm(vert ~ I(year-2012), data = e, family = "binomial")
summary(mod)

vertplot <- data.frame(
  year =  e$year,
  fit = predict(mod, type = "response"),
  lcl = plogis(predict(mod)-2*predict(mod, se.fit = T)$se.fit),
  ucl = plogis(predict(mod)+2*predict(mod, se.fit = T)$se.fit)) %>%
  distinct()

vertplot %>%
  ggplot() +
  geom_point(aes(x = year, y = 100*fit), size = 4) +
  geom_errorbar(aes(x = year, ymin = 100*lcl, ymax = 100*ucl), linewidth = 1, width = 0) +
  scale_x_continuous(breaks = seq(2013, 2023, by = 2)) +
  theme_bw() +
  theme(text = element_text(size = 14)) +
  labs(x = "", y = "% of eDNA impact assessments\ninvolving vertebrates")

# ggsave("figures/figX_vert_eDNA_assess_over_time_full.png", dpi = 400, width = 4, height = 4)
```

# bar charts of % vertebrates and % terrestrial over time
```{r}
# % vertebrates over time
(vert <- e %>%
    # filter(grepl(hab_gen, pattern = "forest") | 
    #          grepl(hab_gen, pattern = "grassland") |
    #          grepl(hab_gen, pattern = "cave") |
    #          grepl(hab_gen, pattern = "other terrestrial")) %>%
  group_by(year) %>%
  summarize(numvert = sum(vert),
            totpubs = length(vert),
            nonvert = totpubs-numvert,
            pctvert = 100*numvert/totpubs,
            .groups = "drop") %>%
  filter(year > 2016) %>%
  ggplot() +
  geom_col(aes(x = year, y = pctvert)) +
  geom_rect(aes(xmin = 2012.55, xmax = 2016.45, ymin = 0, ymax = 6.67), fill = "gray35") +
  geom_text(aes(x = year, y = 0.7, label = totpubs), color = "darkgray", size = 3) +
  geom_text(aes(x = 2014.5, y = 0.7, label = "n = 15"), color = "darkgray", size = 3) +
  theme_bw() +
  theme(text = element_text(size = 14)) +
  labs(x = "", y = "% of eDNA impact assessments\ninvolving vertebrates")
)

# % terrestrial over time
(terr <- e %>%
  group_by(year) %>%
  summarize(numterr = sum(terr),
            totpubs = length(terr),
            nonvert = totpubs-numterr,
            pctvert = 100*numterr/totpubs,
            .groups = "drop") %>%
  filter(year > 2016) %>%
  ggplot() +
  geom_col(aes(x = year, y = pctvert)) +
  geom_rect(aes(xmin = 2012.55, xmax = 2016.45, ymin = 0, ymax = 46.67), fill = "gray35") +
  geom_text(aes(x = year, y = 2.4, label = totpubs), color = "darkgray", size = 3) +
  geom_text(aes(x = 2014.5, y = 2.4, label = "n = 15"), color = "darkgray", size = 3) +
  theme_bw() +
  theme(text = element_text(size = 14)) +
  labs(x = "", y = "% of eDNA impact assessments\nin terrestrial habitats")
)

gg <- gridExtra::arrangeGrob(vert, terr, ncol = 2)
  
plot(gg)

ggsave("figures/fig3_pct_vert_terr_over_time2_full.png", plot = gg, dpi = 400, width = 8, height = 5)

```

# number of taxa
```{r}

b <- e %>%
  select(numtaxgroups) %>%
  mutate(type = "eDNA assessment") %>%
  bind_rows(mutate(select(p, numtaxgroups), type = "FCO projects"))

b %>%
  # filter(numtaxgroups>0) %>%
  ggplot() +
  geom_boxplot(aes(x = type, y = numtaxgroups, 
                   color = type), outlier.alpha = 0) +
  geom_point(aes(x = jitter(as.numeric(as.factor(type)), 0.3), y = numtaxgroups,
                 color = type), 
             shape = 21, alpha = 0.75, size = 2.5) +
  scale_color_manual(values = c("firebrick", "steelblue")) +
  scale_y_continuous(limits = c(0,7)) +
  labs(x = "", y = "No. taxonomic groups monitored") +
  theme_bw() +
  theme(text = element_text(size = 11)) +
  guides(color = "none")

ggsave("figures/fig4_num_taxa_monitored2_full.png", width = 3, height = 3, dpi =400)
```

# number of taxa vs. number of methods
```{r}

b <- e %>%
  select(numtaxgroups) %>%
  mutate(type = "eDNA assessment",
         nummethods = 1) %>%
  bind_rows(mutate(select(p, numtaxgroups, nummethods), 
                   type = "FCO projects",
                   nummethods = nummethods+1))

b %>%
  filter(type == "FCO projects") %>%
  ggplot() +
  geom_boxplot(aes(x = as.factor(nummethods), y = numtaxgroups, 
                   color = type),
             position = position_dodge(width = 1),
             outlier.alpha = 0) +
    geom_point(aes(x = jitter(as.numeric(as.factor(nummethods)), 0.3)+1, y = numtaxgroups,
                 color = type), 
             shape = 21, alpha = 0.75, size = 3,
             data = filter(b, type == "FCO projects", nummethods>1)) +
    geom_point(aes(x = jitter(as.numeric(as.factor(nummethods)), 1.3), y = numtaxgroups,
                 color = type), 
             shape = 21, alpha = 0.75, size = 3,
             data = filter(b, type == "FCO projects", nummethods==1)) +
  # note: geom_points are split up like this in case showing both eDNA/FCO is desired
    # geom_point(aes(x = jitter(as.numeric(as.factor(nummethods))-0.25, 2.3), y = numtaxgroups,
    #              color = type), 
    #          shape = 21, alpha = 0.75, size = 3,
    #          data = filter(b, type == "eDNA assessment")) +
  scale_color_manual(values = c("olivedrab", "darkorange")) +
  scale_y_continuous(limits = c(0,7), breaks = 1:7) +
  labs(x = "No. monitoring methods", y = "No. taxonomic groups") +
  theme_classic() +
  theme(text = element_text(size = 11)) +
  guides(color = "none") +
  theme(text = element_text(size = 14))

ggsave("figures/fig_taxa_monitored_v_methods_used.png", width = 3, height = 3, dpi =400)
```

# number of taxa vs. number of primer regions
```{r}

b <- e %>%
  select(numtaxgroups, numregions, method) %>%
  mutate(type = "eDNA assessment",
         nummethods = 1) %>%
  filter(!is.na(numregions) & numregions != "NA") %>%
  filter(grepl(method, pattern = "metabar"))

b %>%
  ggplot() +
  geom_boxplot(aes(x = as.factor(numregions), y = numtaxgroups),
             outlier.alpha = 0, color = "firebrick") +
    geom_point(aes(x = jitter(as.numeric(as.factor(numregions)), 0.3), y = numtaxgroups),
                 color = "firebrick", 
             shape = 21, alpha = 0.75, size = 3) +
  scale_y_continuous(limits = c(0,7), breaks = 1:7) +
  labs(x = "No. genomic regions", y = "No. taxonomic groups") +
  theme_classic() +
  theme(text = element_text(size = 11)) +
  guides(color = "none") +
  theme(text = element_text(size = 14))

ggsave("figures/fig_taxa_monitored_v_markers_used.png", width = 3, height = 3, dpi =400)
```
# tally stats for results section
```{r}
# % stats for forest carbon project reports

# where are the projects
table(read_xlsx("data/FCO_biodiversity_data_working.xlsx", sheet = "FCO_projects")$Region)/82
   #     Africa          Asia Latin America 
   # 0.09756098    0.50000000    0.40243902 
   
# no. projects with some form of formal monitoring
(numanim <- p %>% 
  filter(numanimalgroups>0) %>%
  nrow())
numanim/nrow(p) # 85

sum(p$birds)/nrow(p)
sum(p$mammals)/nrow(p)
sum(p$reptiles)/nrow(p)
sum(p$amphibians)/nrow(p)
sum(p$invertebrates)/nrow(p)
sum(p$fish)/nrow(p)
sum(p$numanimalgroups==0) # no animal monitoring
sum(p$numanimalgroups==0)/nrow(p) # % no animal monitoring
sum(p$plants)/nrow(p) # some plant monitoring mentioned
(someformal <- p %>%
  filter(formal_obs != 0 | accous != 0 | camtrap != 0) %>%
  nrow()) # some plot-based animal monitoring including camera trapping or traps
someformal/nrow(p)
(sum(p$formal_obs != 0)) # visual animal monitoring 
(sum(p$incidental_obs != 0)) # camera trap animal monitoring 
(sum(p$camtrap != 0)) # camera trap animal monitoring 
(sum(p$traps != 0)) # trap-based animal monitoring 
(sum(p$accous != 0)) # acoustic animal monitoring 
(sum(p$DNA != 0)) # DNA animal monitoring 
(sum(p$bg == 1)) # 27 gold projects
nrow(p) - (sum(p$bg == 1)) # 53 non-gold
# number of Biodiversity Gold projects with monitoring tech
p %>%
  filter(bg == 1) %>%
  filter(camtrap != 0 | DNA != 0 | accous != 0) %>%
  nrow()
# number of formal plot-based monitoring that reported sample size
tmp <- p %>%
  filter(formal_obs != 0 | accous != 0 | camtrap != 0 | traps != 0) %>%
  mutate(formal_obs = as.numeric(formal_obs),
         accous = as.numeric(accous),
         camtrap = as.numeric(camtrap),
         traps = as.numeric(traps)) %>%
  rowwise() %>%
  mutate(sampsum = sum(c_across(c(formal_obs, accous, camtrap, traps)), na.rm = T)) %>%
  ungroup() %>%
  filter(sampsum > 0)
# mean sample sizes where reported
mean(as.numeric(p$camtrap[as.numeric(p$camtrap)>0]), na.rm = T)
mean(as.numeric(p$formal_obs[as.numeric(p$formal_obs)>0]), na.rm = T)
mean(as.numeric(p$traps[as.numeric(p$traps)>0]), na.rm = T)

# % stats for eDNA assessment literature
sum(e$birds)/nrow(e)
sum(e$mammals)/nrow(e)
sum(e$reptiles)/nrow(e)
sum(e$amphibians)/nrow(e)
sum(e$invertebrates)/nrow(e)
sum(e$fish)/nrow(e)
sum(e$prokaryotes)/nrow(e)
sum(e$fungi)/nrow(e)
sum(e$protists)/nrow(e)
table(e$hab_gen)
# how many terrestrial
(eterr <- e %>%
  filter(grepl("forest", hab_gen) |
         grepl("grassland", hab_gen) |
         grepl("terrestrial", hab_gen) |
         grepl("cave", hab_gen)) %>%
  tally()) # 134
eterr/nrow(e)
# how many studies in each habitat
table(e$hab_gen)
# what taxa monitored in forests
ef <- e %>% 
  filter(grepl("forest", hab_gen))
sum(ef$birds)
sum(ef$mammals)
sum(ef$reptiles)
sum(ef$amphibians)
sum(ef$invertebrates)
sum(ef$fish)
sum(ef$prokaryotes)
sum(ef$fungi)
sum(ef$protists)
# how many studies that inolved forest were in each impact category
e %>% 
  filter(grepl("forest", hab_gen)) %>%
  group_by(impact) %>%
  tally()

# how many in each substrate category
e %>% 
  filter(grepl("forest", hab_gen)) %>%
  group_by(substrate) %>%
  tally()

# where are eDNA studies
table(e$region)
table(e$country_area)

# average sample size of eDNA studies
mean(e$numsamp, na.rm = T)
sd(e$numsamp, na.rm = T)
summary(e$numsamp, na.rm = T)
sd(e$numsamp, na.rm = T) / sqrt(length(e[!is.na(e$numsamp),]$numsamp))
mean(e$numsamp, na.rm = T)


```
# Map the CCB projects
```{r}

library(sf)
library(rnaturalearth)
'%notin%' <- Negate('%in%')

world <- rnaturalearth::ne_countries(returnclass = "sf")
plot(world)

# summarize count of projects by country, fixing name of Tanzania
pcountry <- p %>%
  group_by(country_area) %>%
  summarize(count = length(country_area)) %>%
  mutate(sovereignt = case_when(country_area == "Tanzania" ~ "United Republic of Tanzania",
                                TRUE ~ country_area))

worldp <- world %>%
  left_join(pcountry, by = c("sovereignt"))

# check which ones matched correctly (out of 17)
test = worldp %>% filter(!is.na(count))

# make lat/lon into sf object
psf <- st_as_sf(p[,c("lon_approx", "lat_approx")],
                coords = c("lon_approx", "lat_approx"))

# plot
psf <- p %>%
  filter(!is.na(lat_approx)) %>%
  select(ID, lat_approx, lon_approx) %>%
  st_as_sf(coords = c("lon_approx", "lat_approx"),
           crs = 4326)

# Deactivate s2
sf::sf_use_s2(FALSE)

worldp %>%
ggplot +
  geom_sf(aes(fill = count)) +
  geom_sf(aes(), 
             shape = "x", data = psf, color = "red") +
  scale_fill_viridis_b() +
  theme_bw() +
  labs(x = "", y = "", fill = "No.\nprojects") +  
coord_sf(crs = st_crs("ESRI:54030"))

ggsave("figures/fig_map_projects3.png", width = 6, height = 6, dpi =400)

# summarize number of CCB projects at different latitudes
sum(abs(as.numeric(p$lat_approx))<23.5, na.rm = T) # 33
sum(abs(as.numeric(p$lat_approx))>23.5, na.rm = T) # 34
sum(abs(as.numeric(p$lat_approx))<35, na.rm = T) # 54
sum(abs(as.numeric(p$lat_approx))>35, na.rm = T) # 13

sum(abs(as.numeric(p$lat_approx))<35, na.rm = T) # 54
sum(abs(as.numeric(p$lat_approx))>35, na.rm = T) # 13


sum(pall$`trop-subtrop`==1, na.rm = T) # 85
sum(!is.na(pall$`trop-subtrop`)) # 100
mean(abs(as.numeric(pall[pall$`trop-subtrop`!=1,]$lat_approx)), na.rm = T)

sum(!is.na(p$lat_approx)) # 67
48/56
8/56




```

# Map the eDNA assessment literature
```{r}

library(sf)
library(rnaturalearth)
'%notin%' <- Negate('%in%')

world <- rnaturalearth::ne_countries(returnclass = "sf")
plot(world)

# summarize count of projects by country, fixing name of Tanzania
ecountry <- e %>%
  mutate(admin = case_when(country_area == "Tanzania" ~ "United Republic of Tanzania",
                                country_area == "USA" ~ "United States of America",
                                country_area == "Hawaii (USA)" ~ "United States of America",
                                country_area == "Hawaii" ~ "United States of America",
                                country_area == "Mayotte (France)" ~ "France",
                                country_area == "UK" ~ "United Kingdom", 
                                country_area == "Czech Republic" ~ "Czechia", 
                                country_area == "Curacao" ~ "Netherlands",
                                TRUE ~ country_area)) %>%
  filter(country_area %notin% c("Amazon basin", "Atlantic Ocean",
                                "Bay of Biscay", "S. Atlantic / Antarctic Ocean",
                                "Mediterranean Sea", "Mekong River",
                                "Multiple", "South Pacific")) %>%
  group_by(admin) %>%
  summarize(count = length(admin))

worlde <- world %>%
  left_join(ecountry, by = c("admin"))

# check which ones matched correctly (out of 17)
test = worlde %>% filter(!is.na(count))
ecountry[ecountry$admin %notin% test$admin,]$admin

# plot
esf <- e1 %>%
  filter(!is.na(lat_approx)) %>%
  select(paperid, lat_approx, lon_approx) %>%
  st_as_sf(coords = c("lon_approx", "lat_approx"),
           crs = 4326)

# Deactivate s2
sf::sf_use_s2(FALSE)

worlde %>%
ggplot +
  geom_sf(aes(fill = count)) +
  geom_sf(aes(), 
             shape = "x", data = esf, color = "red") +
  scale_fill_viridis_b() +
  theme_bw() +
  labs(x = "", y = "", fill = "No.\nstudies") +  
coord_sf(crs = st_crs("ESRI:54030"))

ggsave("figures/fig_map_eDNAstudies.png", width = 6, height = 6, dpi =400)

# summarize number of projects at different latitudes
sum(abs(e1$lat_approx)<=23.5, na.rm = T) # 18
sum(abs(e1$lat_approx)>23.5, na.rm = T) # 66
sum(abs(e1$lat_approx)<=35, na.rm = T) # 40
sum(abs(e1$lat_approx)>35, na.rm = T) # 44

sum(!is.na(e1$lat_approx)) # 84
40/84

# just terrestrial studies
e.terr <- e1 %>%
    filter(grepl(hab_gen, pattern = "forest") |
             grepl(hab_gen, pattern = "grassland") |
             grepl(hab_gen, pattern = "cave") |
             grepl(hab_gen, pattern = "other terrestrial"))

sum(abs(e.terr$lat_approx)<=35, na.rm = T) # 19
sum(abs(e.terr$lat_approx)>35, na.rm = T) # 20
sum(!is.na(e.terr$lat_approx)) # 39
19/39

```
